Extend/Order.php

```diff
@@ -26,45 +33,161 @@ class Order extends Order_parent
     const STATE_CANCELED = 'canceled';
     const STATE_REFUNDED = 'refunded';
 
-    private $_aModulePaymentTypes = array();
+    /**
+     * Returns an array of available states.
+     *
+     * @return array
+     */
+    public static function getStates(): array
+    {
+        return [
+            self::STATE_PENDING,
+            self::STATE_AUTHORIZED,
+            self::STATE_PROCESSING,
+            self::STATE_CANCELED,
+            self::STATE_REFUNDED,
+        ];
+    }
 
     /**
-     * Order constructor.
+     * Returns the country associated with the order billing address.
      *
-     * @SuppressWarnings(PHPMD.Coverage)
+     * @return Country
      */
-    public function __construct()
+    public function getOrderBillingCountry(): Country
     {
-        parent::__construct();
+        $oCountry = oxNew(Country::class);
+        $oCountry->load($this->oxorder__oxbillcountryid->value);
 
-        $this->_aModulePaymentTypes[] = Paypal_Payment_Method::getName(true);
+        return $oCountry;
     }
 
     /**
-     * Checks if the Paymenttype is one of the module's
+     * Returns the country associated with the order shipping address.
      *
-     * @return bool
+     * @return Country
+     */
+    public function getOrderShippingCountry(): Country
+    {
+        $oCountry = oxNew(Country::class);
+        $oCountry->load($this->oxorder__oxdelcountryid->value);
+
+        return $oCountry;
+    }
+
+    /**
+     * Returns the payment associated with the order.
      *
-     * @SuppressWarnings(PHPMD.Coverage)
+     * @return Payment
      */
-    public function isModulePaymentType(): bool
+    public function getOrderPayment(): Payment
     {
-        return in_array($this->getPaymentType()->oxuserpayments__oxpaymentsid->value, $this->_aModulePaymentTypes);
+        $oPayment = oxNew(Payment::class);
```

Comment:
Parent Order class already has $this->_oPayment. Could be checked and returned if not empty, otherwise stored into?

Reply:
I thought of that too, but `_oPayment` is of the type `UserPayment`, not `Payment` (for whatever reason…).